### ISSUE TITLE
[block-producer]: redesign tracking issue

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [0xPolygonMiden/miden-node#514]
The block-producer redesign feature is being developed against the `next-block-producer` branch until it is ready to merge into `next` proper.

This issue tracks the remaining work after #502, initially derived from [this comment](https://github.com/0xPolygonMiden/miden-node/pull/502#issuecomment-2393925630).

- [x] input and output note state tracking
- [x] data propagation from mempool to batch and block builders
- [x] batch builder with [configurable failure/work rate](https://github.com/0xPolygonMiden/miden-node/pull/502#discussion_r1789644365)
- [x] block builder with configurable failure/work rate
- [x] tests
- [x] replacing existing framework
- [x] #508 
- [x] #519 
- [x] #531

**Comment by bobbinth** [comment]
@Mirko-von-Leipzig - should we close this issue?
